catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AuthenticationApplication.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.security;
+
+import static spark.Spark.post;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.codice.ddf.security.rest.authentication.service.AuthenticationService;
+import spark.servlet.SparkApplication;
+
+public class AuthenticationApplication implements SparkApplication {
+
+  private AuthenticationService authenticationService;
+
+  public AuthenticationApplication(AuthenticationService authenticationService) {
+    this.authenticationService = authenticationService;
+  }
+
+  @Override
+  public void init() {
+    post(
+        "/login",
+        (req, res) -> {
+          Map<String, String> formParams = new HashMap<>();
+
+          // Extract form parameters
+          String body = req.body();
+          Pattern regExp =
+              Pattern.compile(
+                  "^Content-Disposition: form-data; name=\"(.+)\"$[\r\n]+(.+)$", Pattern.MULTILINE);
```

Comment:
❓ I'm unclear about the use of this regex. Is this the standard/approved way of extracting multipart data with Spark? It seems rather brittle.

Reply:
Looks we can do stuff like this. Do any of these not work @blen-desta ?

https://stackoverflow.com/questions/29373468/spark-java-how-to-handle-multipart-form-data-input

https://www.firstfewlines.com/post/spark-java-file-upload-example/